### DISCUSSION TITLE
Error message "Context was requested but was not available"

### DISCUSSION THREAD

**Discussion by acharseth** [deployphp/deployer#3706] [Help needed]
I would like to do something like this in deploy.php:

```php
<?php
namespace Deployer;

require 'recipe/laravel.php';
require 'contrib/rsync.php';

// Config
$host=currentHost()->getAlias();

// Common configurations:
set('rsync_src', "/tmp/myApp/$host");
...
```

But the `$host=currentHost()->getAlias();` causes an error message: `Context was requested but was not available`.

How can I use the host-parameter in a `set`?

**Comment by antonmedv** [comment]
currentHost() only defined in tasks.

**Reply by acharseth** [reply]
Some other way I can get the host parameter?
I tried `$host=$argv[2]` but then get an error message saying that $argv is not defined.
